source/CMakeLists.txt

```diff
@@ -130,6 +130,71 @@ set(XLNT_SOURCES ${CELL_SOURCES} ${CHARTS_SOURCES} ${CHARTSHEET_SOURCES}
   ${WORKSHEET_SOURCES} ${DETAIL_SOURCES} ${DETAIL_CRYPTO_SOURCES}
   ${DRAWING_SOURCES} ${MINIZ_SOURCES})
 
+# ============================================================================
+# ZIP Backend Configuration (MUST be before add_library to affect GLOB files)
+# ============================================================================
+# xlnt supports two ZIP compression backends:
+#
+# 1. "builtin" (default):
+#    - Uses miniz library (embedded in xlnt)
+#    - Lightweight, no external dependencies
+#    - Does NOT support Zip64 (files > 4GB)
+#    - Recommended for most use cases
+#
+# 2. "minizipng":
+#    - Uses minizip-ng library with zlib-ng (auto-fetched via CMake)
+#    - Full Zip64 support (handles large files and archives)
+#    - Requires external dependencies (minizip-ng, zlib-ng)
+#    - Recommended for large Excel files or advanced ZIP features
+#
+# To change the backend, set XLNT_ZIP_BACKEND to "builtin" or "minizipng"
+# Example: cmake -DXLNT_ZIP_BACKEND=minizipng ..
+# ============================================================================
+
+set(XLNT_ZIP_BACKEND "builtin" CACHE STRING "ZIP backend: builtin or minizipng")
+set_property(CACHE XLNT_ZIP_BACKEND PROPERTY STRINGS builtin minizipng)
+
+if(XLNT_ZIP_BACKEND STREQUAL "minizipng")
```

Comment:
I'm not against implementing this logic to automatically built the minizipng dependency. An alternative is to expect the user to built the library himself if he want to use the non-default backend. Or use a variable similar to "MZ_FETCH_LIBS", to allow the user the enable/disable the automatic building of the dependencies. This allows the user more fine grained control over the built options and built version of the library (if needed).